llvm/lib/TableGen/TGLexer.cpp

```diff
@@ -1055,20 +1048,16 @@ bool TGLexer::prepSkipDirectiveEnd() {
 }
 
 bool TGLexer::prepIsProcessingEnabled() {
-  for (const PreprocessorControlDesc &I :
-       llvm::reverse(*PrepIncludeStack.back()))
```

Comment:
I have no idea why it was using `llvm::reverse` here. Maybe the last element usually has a higher chance to bail out?

Reply:
I did think about that, but really if you have nested #ifs I don't see any reason why the inner (or outer) condition would have a higher chance of being false.